nautobot/core/models/utils.py

```diff
@@ -203,3 +205,24 @@ def deconstruct_composite_key(composite_key):
     values = [unquote_plus(value) for value in composite_key.split(COMPOSITE_KEY_SEPARATOR)]
     values = [value if value != "\0" else None for value in values]
     return values
+
+
+def construct_natural_slug(values):
+    """
+    Convert the given list of natural key values to a single human-readable string.
+
+    A third-party lossy `slugify()` function is used to convert each natural key value to a
+    slug, and then they are joined with an underscore.
+
+    - Spaces or repeated dashes are converted to single dashes.
+    - Remove characters that are not alphanumerics, underscores, or hyphens.
```

Comment:
This is more nuanced now that we're using `python-slugify` right? Non-ASCII characters are converted to closest-match alphanumeric characters now?

Reply:
Yes. 

```python
In [2]: w = "Über Ålpha"

In [3]: slugify(w)
Out[3]: 'uber-alpha'
```

Reply:
Will also point out https://github.com/nautobot/nautobot/issues/4368. Sounds like this will reasonably solve that ask as it relates to natural slug, at least for now.

Reply:
> Will also point out #4368. Sounds like this will reasonably solve that ask as it relates to natural slug, at least for now.

Yes. The revised slugify pattern is aware of accents/ligatures, so the example in #4368 would be:

```python
'device-rack-espana'
```